unit/unit_test.go

```diff
@@ -472,3 +472,38 @@ func BenchmarkMulMapLargeSame(b *testing.B) {
 		u2.Mul(u1)
 	}
 }
+
+func BenchmarkDimensionsString(b *testing.B) {
+	oneDim := Dimensions{CurrentDim: 1}
+	threeDims := Dimensions{
+		CurrentDim:           1,
+		LengthDim:            2,
+		LuminousIntensityDim: 3,
+	}
+	allDims := Dimensions{
+		CurrentDim:           1,
+		LengthDim:            2,
+		LuminousIntensityDim: 3,
+		MassDim:              4,
+		MoleDim:              5,
+		TemperatureDim:       6,
+		TimeDim:              7,
+		AngleDim:             8,
+	}
+
+	b.Run("oneDim", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			_ = oneDim.String()
```

Comment:
The assignment to `_` is a no-op here. It does not ensure that the `String` method call is not optimised away. So please remove it. If you are worried about the optimisation occurring, please add a cheap conditional (say `!= ""`) with a `b.FailNow()` in the fail case to elevate the value to an observable effect.

Reply:
Thanks for pointing this out. Actually I was already aware that this doesn't stop the method call from being optimised away, it was more of a habit to make it super clear that we're intentionally ignoring the result.

But it sounds sensible to use `b.FailNow()` anyway, so I've just force-pushed a change to apply this.